File: src/otherAlgorithms/PartialSum.java
package otherAlgorithms;

import java.io.*;

public class PartialSum {
    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] ns = br.readLine().split(" ");
        int n = Integer.parseInt(ns[0]), s = Integer.parseInt(ns[1]);
        int[] arr = new int[n], sumArr = new int[n];

        String[] in = br.readLine().split(" ");
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(in[i]);
        }

        sumArr[0] = arr[0];

        for(int i = 1; i < n; i++){
            sumArr[i] = sumArr[i - 1] + arr[i];
        }

        int start = 0, end = 0, minLen = Integer.MAX_VALUE;

        while(start < n && end < n){
            int sum = sumArr[end] - sumArr[start] + arr[start];

            if(sum < s){
                end++;
            }else{
                minLen = Math.min(minLen, end - start + 1);
                start++;
            }
        }

        String result = "";

        if(minLen == Integer.MAX_VALUE){
            result = "0";
        }else{
            result = String.valueOf(minLen);
        }

        bw.write(result);
        bw.flush();
        bw.close();
    }
}

File: JS/otherAlgorithms/partialSum.js
"use strict";

const fs = require("fs");

const input =
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString().trim().split("\n")
    : `10 15
5 1 3 5 10 7 4 9 2 8`.split("\n");

const [n, s] = input[0].split(" ").map((num) => Number(num));
const arr = input[1].split(" ").map((num) => Number(num));
const sumArr = Array(n).fill(0);

sumArr[0] = arr[0];

for (let i = 1; i < n; i++) {
  sumArr[i] = sumArr[i - 1] + arr[i];
}

let start = 0;
let end = 0;

let minLen = Infinity;

while (start < n && end < n) {
  const partialSum = sumArr[end] - sumArr[start] + arr[start];

  if (partialSum < s) {
    end++;
  } else {
    minLen = Math.min(minLen, end - start + 1);
    start++;
  }
}

console.log(minLen === Infinity ? 0 : minLen);
